js/packages/teams-ai/src/TaskModules.spec.ts

```diff
@@ -0,0 +1,194 @@
+            };
+
+            mockApp.taskModules.fetch(badRouteSelector, async (_context, _state, _data) => {
+                return 'testFailed'; // the test should not reach this point
+            });
+            await assert.rejects(async () => {
+                await adapter.processActivity(testActivity, async (context) => {
+                    await mockApp.run(context);
+                });
+            });
+        });
+        it('fetch() with custom RouteSelector unhappy path', async () => {
+            const testActivity = { channelId: Channels.Msteams, type: ActivityTypes.Invoke, name: 'incorrectName' };
+            const badRouteSelector = (context: TurnContext) => {
+                return Promise.resolve(true);
+            };
+
+            mockApp.taskModules.fetch(badRouteSelector, async (context, _state, _data) => {
+                return Promise.resolve('');
+            });
+            await assert.rejects(async () => {
+                await adapter.processActivity(testActivity, async (context) => {
+                    await mockApp.run(context);
+                });
+            });
+        });
+
+        // it('fetch() with array of verbs', async () => {});
+    });
+
+    describe(SUBMIT_INVOKE_NAME, () => {
+        it('submit() with custom RouteSelector happy path', async () => {
+            const expectedVerbValue = 'verbValue';
+            const testActivity = createTestInvoke(SUBMIT_INVOKE_NAME, {
+                data: { verb: expectedVerbValue }
+            });
+            testActivity.channelId = Channels.Msteams;
+            const messageResult = 'messageResult';
+            mockApp.taskModules.submit(expectedVerbValue, async (_context, _state, _data) => {
+                return messageResult;
+            });
+
+            await adapter.processActivity(testActivity, async (context) => {
+                await mockApp.run(context);
+                const response = context.turnState.get(INVOKE_RESPONSE_KEY);
+                assert.deepEqual(response.value, {
+                    status: 200,
+                    body: { task: { type: 'message', value: messageResult } }
+                });
+            });
+        });
+
+        it('submit() with custom RouteSelector unhappy path', async () => {
+            const testActivity = { channelId: Channels.Msteams, type: ActivityTypes.Invoke, name: 'incorrectName' };
+            const badRouteSelector = (context: TurnContext) => {
+                return Promise.resolve(true);
+            };
+
+            mockApp.taskModules.submit(badRouteSelector, async (_context, _state, _data) => {
+                return Promise.resolve(undefined);
+            });
+            await assert.rejects(async () => {
+                await adapter.processActivity(testActivity, async (context) => {
+                    await mockApp.run(context);
+                });
+            });
+        });
+
+        // it('submit() with array of verbs', async () => {});
```

Comment:
these might be leftovers from debugging?

Reply:
wooops thanks!